### PR TITLE
[libra-dev] Adding core client library crate

### PR DESCRIPTION
**Pull request by sunmilee**
## Motivation

This PR takes the first step in adding a core client library by creating a libra-dev crate with an initial deserialize account resource class

## Test Plan

tbd

### CODE REVIEW

File: libra-dev/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+repository = "https://github.com/libra/libra"
+homepage = "https://libra.org"
+license = "Apache-2.0"
+publish = false
+edition = "2018"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+libra-types = { path = "../types", version = "0.1.0" }
```
Comment (thefallentree) [inline libra-dev/Cargo.toml]: remove the version part

File: libra-dev/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+lcs = { path = "../common/lcs", version = "0.1.0", package = "libra-canonical-serialization" }
```
Comment (thefallentree) [inline libra-dev/Cargo.toml]: same

File: Cargo.toml [hunk 3]
```diff
@@ -56,6 +56,7 @@ members = [
     "language/vm/vm-runtime/vm-cache-map",
     "language/vm/vm-runtime/vm-runtime-types",
     "language/vm/vm-genesis",
+    "libra-dev",
```
Comment (thefallentree) [inline Cargo.toml]: should we make libra-dev its own workspace ? That way we can explicitly control all the dependencies.

@bmwill  for comment

Comment (bmwill) [reply Cargo.toml]: Making it its own workspace has no impact on controlling the deps, its still trivially easy to provide a path dep. So people do not move it into another workspace.

File: libra-dev/main.c [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	*out = malloc(len);
+	memset(*out, 0 , len);
+	for (i=0; i<len; i++) {
+		if (!hexchr2bin(hex[i*2], &b1) || !hexchr2bin(hex[i*2+1], &b2)) {
+			return 0;
+		}
+		(*out)[i] = (b1 << 4) | b2;
+	}
+	return len;
+}
```
Comment (bmwill) [inline libra-dev/main.c]: This isn't really necessary, I would prefer if we didn't include this hextobin logic and instead just have an array with the contents already in binary form.

File: libra-dev/src/account_resource.rs [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+use libra_types::{
+    account_config::get_account_resource_or_default,
+    account_state_blob::AccountStateBlob,
+    //    byte_array::ByteArray,
+};
+use std::slice;
+
+#[derive(Debug)]
+pub struct DevAccountResource {
```
Comment (bmwill) [inline libra-dev/src/account_resource.rs]: We probably don't need two copies of the struct, the C one should be all we need.

File: types/src/event.rs [hunk 6]
```diff
@@ -40,7 +40,7 @@ impl EventKey {
         EventKey::try_from(HashValue::random().to_vec().as_slice()).unwrap()
     }
 
-    /// Create a unique handle by using an AccountAddress and a counter.
+    /// Create a unique handle by using an AccㄹountAddress and a counter.
```
Comment (bmwill) [inline types/src/event.rs]: don't think this was intentional?

File: libra-dev/main.c (hunk 4 above)
Comment (sunmilee) [reply libra-dev/main.c]: Yeah this was just for testing!

**Comment by sunmilee** [comment]
Closing since we're moving to a new repo

**Pull request closed without merging**